### ISSUE TITLE
Some confusions about the supported token dimension.

### ISSUE THREAD

**Issue by AoboLiang00208** [OpenNLPLab/lightning-attention#15]
Hi! I got some confusions on the following snippets:
https://github.com/OpenNLPLab/lightning-attention/blob/d7439519541e966084eeaaf3ffd63eecc216f414/lightning_attn/ops/lightning_attn_interface.py#L27-L37

If d is larger than 128, take `d=180` or any number for example, the result of `64%d` will always be 64. So I'm confused about these if-else judgements. Won't `m` always be `64` if `d>128`?

Another thing is that, if `d=128`, the lightning_attn function could still work well, so why not set the limit for `m` to 128?

Further, may the function of `is_support` fail under certain circumstances? For example, `dim=5` and `16%5` is not 0. (Or, did I misunderstand the purpose of this function?)

**Comment by Doraemonzzz** [comment]
Thank you for your question. I reviewed the code, and it does appear to be a bug. I will address this in a future fix. What I intended to implement here is splitting the `head_dim` when it is relatively large, such as dividing 256 into two 128s. Therefore, the correct code should be `if d % 64 == 0:`. In most cases, we rarely use a `head_dim` larger than 128, so this bug is unlikely to be triggered under typical circumstances.
